elastic/apm-server
# sampling: clarify storage limit reached error

## Motivation/summary

When tail-based sampling fails due to the storage limit being reached, the error message looks something like this:

    flush pending writes: configured storage limit reached: no space left on device

This implies the disk is actually out of space, but the "no space left on device" is just an artefact of wrapping `syscall.ENOSPC` in `ErrLimitReached`. We remove the ENOSPC to avoid this confusion. Also, we now include the current badger DB size vs. stoarge limit in the error message, and add a unit test.

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Configure TBS, and a storage limit
2. Run some workload until the storage limit is reached
3. Check the logged error messages, ensure that they don't say "no space left on device"

## Related issues

None
